pkg/spireentry/entries.go

```diff
@@ -160,6 +166,23 @@ func renderDNSName(tmpl *template.Template, data *templateData) (string, error)
 	return rendered, nil
 }
 
+func addServiceDNSNames(dnsNames []string, dnsNamesSet map[string]struct{}, endpointsList *corev1.EndpointsList, clusterDomain string) []string {
```

Comment:
what do you think about this function returning the list of dns from endpoints (only) and sorted.
and in the caller add the new dns to the list?
```
dsnNames = append(dnsNames, dnsNamesFromEndpoints( dnsNamesSet, endpointsList, clusterDomain))
``` 

that wil simplify this code

Reply:
Change made.